### PR TITLE
Pyqt fixes

### PR DESCRIPTION
**Pull request by RobertRosca**
While working on the centre position optimisation feature I ran into the error:

```python
AttributeError: module 'PyQt5.QtCore' has no attribute 'Slot'
```

Found [this page](https://stackoverflow.com/questions/50459672/attributeerror-module-pyqt5-qtcore-has-no-attribute-slot) which said that the `Slot` was replaced with `pyqtSlot`, so I changed it and everything ran fine afterwards.

I assume this might be due to PyQt updating, so in addition to changing the calls to Slot, I also pinned the versions of PyQt to under the next major release, which should help prevent these kind of deprecation problems.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def find_files(directory):
           'pyqtgraph',
           'ipywidgets',
           'pyFai',
-          'PyQt5',
-          'PyQt5-sip',
+          'PyQt5 >= 5.13.2, <= 6.0.0',
+          'PyQt5-sip >= 12.7.0, <= 13.0.0',
```
Comment (zhujun98) [inline setup.py]: PyQt5-sip is a dependency of PyQt5.

**Comment by antarcticrainforest** [comment]
Sorry dude, that's a bummer. I had fixed this already but apparently never pushed it before it got merged.

**Comment by zhujun98** [comment]
LGTM

**Comment by antarcticrainforest** [comment]
Anyhoo, sorry for wasting your time. LGTM

**Comment by takluyver** [comment]
Thanks everyone!

We might consider using QtPy, a wrapper maintained by the Spyder developers, which allows compatibility between PyQt and PySide (different bindings for Qt - there's history concerning their licenses). But I think it's a good step to make sure it runs properly under one wrapper first.

**Comment by antarcticrainforest** [comment]
Interesting! Shall we do that in this PR though?

**Comment by takluyver** [comment]
I'd say merge this PR as is - I know Robert has another PR coming that will require more detailed review. We can open an issue for using QtPy later.

**Comment by RobertRosca** [comment]
Sounds good to me, cheers.

p.s. no problem at all Martin, all it took was like 30 seconds of googling and a find-and-replace command =P

**Comment by takluyver** [comment]
Looking at it, pyqtgraph has its own compatibility layer over different Qt bindings:
https://github.com/pyqtgraph/pyqtgraph/blob/develop/pyqtgraph/Qt.py

I suggest that we stick to PyQt5 for now. I'm sure it's possible to support the other bindings, but it's not worth spending time on if everyone who needs it can easily install the one we already support.

**Pull request merged**